lib/fauxpaas/artifact_builder.rb

```diff
@@ -38,5 +44,20 @@ def add_reference(ref, path)
         .write
     end
 
+    # Run the finish_build commands, if they exist. Should any fail,
+    # halt the build and raise an error.
+    def finish_build!(path)
+      steps_path = path/"finish_build.yml"
+      if steps_path.exist?
+        step_list = StepList.new(YAML.safe_load(File.read(steps_path)))
+        Dir.chdir(path) do
+          step_list.steps.each do |step|
+            _, _, status = runner.run(step.cmd)
```

Comment:
We probably want to add `Bundler.with_clean_env do` around `runner.run`, either here or (probably more appropriately) in the runner itself.

Reply:
Good catch. I think this is required. See https://bundler.io/man/bundle-exec.1.html#Shelling-out, especially:

> If you need to shell out to a Ruby command that is not part of your current bundle, use the with_clean_env method with a block.